platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/topology/StaticTopology.java

```diff
@@ -121,4 +115,15 @@ public boolean shouldConnectTo(final NodeId nodeId) {
     public RandomGraph getConnectionGraph() {
         return connectionGraph;
     }
+
+    /**
+     * Returns the index of the given node in the peer list, which must not be the self index
+     *
+     * @param nodeId the node ID
+     * @return the index of the node in the peer list
+     */
+    private int getIndexOfNodeId(@NonNull final NodeId nodeId) {
+        final int index = peerNodes.indexOf(nodeId);
```

Comment:
Instead of iterating the list to find the index, we should instead create a `Map<NodeId, Integer>` for `O(1)` lookup time.

Reply:
O(1) here would be nice but i still have to keep a list of peers for the neighbors. To use a map, the values of the map would be a set of peers which must still be converted to a list in O(n). No good trade-off here.

Reply:
`O(1)` is non-optional for stuff like this.

Reply:
Perhaps we can get `O(1)` with a much similar approach. Is there any reason we can't just compare node IDs directly? The index is really just a proxy for node IDs. All we really want to do is figure out if the other node comes first in the address book, and if you have a lower node ID you will always come first in the address book.

Reply:
resolving, as this is now outdated